massa-node/src/main.rs

```diff
@@ -814,37 +816,41 @@ async fn launch(
     let mut api_config = api_config.clone();
     api_config.enable_ws = false;
 
-    // Whether to spawn gRPC API
-    let grpc_handle = if SETTINGS.grpc.enabled {
-        let grpc_config = GrpcConfig {
-            enabled: SETTINGS.grpc.enabled,
-            accept_http1: SETTINGS.grpc.accept_http1,
-            enable_cors: SETTINGS.grpc.enable_cors,
-            enable_health: SETTINGS.grpc.enable_health,
-            enable_reflection: SETTINGS.grpc.enable_reflection,
-            enable_mtls: SETTINGS.grpc.enable_mtls,
-            bind: SETTINGS.grpc.bind,
-            accept_compressed: SETTINGS.grpc.accept_compressed.clone(),
-            send_compressed: SETTINGS.grpc.send_compressed.clone(),
-            max_decoding_message_size: SETTINGS.grpc.max_decoding_message_size,
-            max_encoding_message_size: SETTINGS.grpc.max_encoding_message_size,
-            concurrency_limit_per_connection: SETTINGS.grpc.concurrency_limit_per_connection,
-            timeout: SETTINGS.grpc.timeout.to_duration(),
-            initial_stream_window_size: SETTINGS.grpc.initial_stream_window_size,
-            initial_connection_window_size: SETTINGS.grpc.initial_connection_window_size,
-            max_concurrent_streams: SETTINGS.grpc.max_concurrent_streams,
-            tcp_keepalive: SETTINGS.grpc.tcp_keepalive.map(|t| t.to_duration()),
-            tcp_nodelay: SETTINGS.grpc.tcp_nodelay,
+    // Whether to spawn gRPC PUBLIC API
+    let grpc_public_handle = if SETTINGS.grpc.public.enabled {
+        let grpc_public_config = GrpcConfig {
+            enabled: SETTINGS.grpc.public.enabled,
```

Comment:
It's not possible to factorize a bit the two config here and in config.toml ?

Reply:
Yes, we could make It less boilerplate

Reply:
Do you want to do it here or in a follow up PR ?

Reply:
done here